ludwig/models/trainer.py

```diff
@@ -744,14 +744,15 @@ def run_evaluation(
                 self.early_stop,
                 self.skip_save_model,
             )
-            if should_break:
-                return should_break
         else:
             # There's no validation, so we save the model.
             if self.is_coordinator() and not self.skip_save_model:
                 torch.save(self.model.state_dict(), model_weights_path)
 
-        return False
+        # Trigger eval end callback after any model weights save for complete checkpoint
```

Comment:
Thanks for adding a comment about why the callback should be after model saving.